edit/harvesterhci.io.virtualmachineimage.vue

```diff
@@ -0,0 +1,268 @@
+import NameNsDescription from '@/components/form/NameNsDescription';
+import RadioGroup from '@/components/form/RadioGroup';
+import LabelValue from '@/components/LabelValue';
+import CreateEditView from '@/mixins/create-edit-view';
+import { _EDIT } from '@/config/query-params';
+import { IMAGE_FILE_FORMAT } from '@/config/constant';
+import { HCI as HCI_ANNOTATIONS } from '@/config/labels-annotations';
+import { exceptionToErrorsArray } from '@/utils/error';
+
+const DOWNLOAD = 'download';
+const UPLOAD = 'upload';
+
+export default {
+  name: 'EditImage',
+
+  components: {
+    Tab,
+    Tabbed,
+    KeyValue,
+    CruResource,
+    LabeledInput,
+    NameNsDescription,
+    RadioGroup,
+    LabelValue,
+  },
+
+  mixins: [CreateEditView],
+
+  props: {
+    value: {
+      type:     Object,
+      required: true,
+    },
+  },
+
+  data() {
+    if ( !this.value.spec ) {
+      this.$set(this.value, 'spec', { sourceType: DOWNLOAD });
+    }
+
+    return {
+      url:         this.value.spec.url,
+      files:       [],
+      displayName: '',
+      resource:    '',
+      headers:     {},
+      fileUrl:     '',
+      file:        '',
+    };
+  },
+
+  computed: {
+    uploadFileName() {
+      return this.file?.name || '';
+    },
+
+    isEdit() {
+      return this.mode === _EDIT;
+    },
+
+    imageName() {
+      return this.value?.metadata?.annotations?.[HCI_ANNOTATIONS.IMAGE_NAME] || '-';
+    },
+
+    isCreateEdit() {
+      return this.isCreate || this.isEdit;
+    },
+  },
+
+  watch: {
+    'value.spec.url'(neu) {
+      const url = neu.trim();
+      const suffixName = url.split('/').pop();
+      const fileSuffiic = suffixName.split('.').pop().toLowerCase();
+
+      this.value.spec.url = url;
+      if (IMAGE_FILE_FORMAT.includes(fileSuffiic)) {
+        if (!this.value.spec.displayName) {
+          this.$refs.nd.changeNameAndNamespace({ text: suffixName });
+        }
+      }
+    },
+
+    'value.spec.displayName'(neu) {
+      this.displayName = neu;
+    },
+
+    'value.spec.sourceType'() {
+      this.$set(this, 'file', null);
+      this.url = '';
+
+      if (this.$refs?.file?.value) {
+        this.$refs.file.value = null;
+      }
+    },
+  },
+
+  methods: {
+    async saveImage(buttonCb) {
+      this.value.metadata.generateName = 'image-';
+
+      if (this.value.spec.sourceType === UPLOAD && this.isCreate) {
+        try {
+          this.value.spec.url = '';
+
+          if (!this.value.metadata.annotations) {
+            this.value.metadata.annotations = {};
+          }
+
+          this.value.metadata.annotations[HCI_ANNOTATIONS.IMAGE_NAME] = this.file?.name;
+
+          const res = await this.value.save({ extend: { isRes: true } });
+
+          const formData = new FormData();
+
+          formData.append('chunk', this.file);
+
+          this.$axios.post(`/v1/harvester/harvesterhci.io.virtualmachineimages/${ res.id }?action=upload&size=${ this.file.size }`, formData, {
```

Comment:
$axios shouldn't be used directly, try dispatching an action via the current store, if you search for `/request'` you should get some good examples.

Think it should also be moved into the model for this resource type (something like `this.value.uploadImage`)

Reply:
I will do it